kolibri/core/assets/src/validators.js

```diff
@@ -25,3 +25,195 @@ export function validateLearningActivity(arr) {
   const isValidLearningActivity = v => Object.values(LearningActivities).includes(v);
   return arr.length > 0 && arr.every(isValidLearningActivity);
 }
+export function objectPropValidator( options ){
+	var args = arguments || [];
+    var _getType = function(variable) {
+        var match = false;
+        var rawType = "";
+        try {
+            match = variable && variable.toString().match(/^\s*function (\w+)/);
+        } catch {}
+        try {
+            rawType = Object.prototype.toString.call(variable).slice(8, -1);
+        } catch {}
+        return match ? match[1] : (rawType.length > 0 ? rawType : ((typeof variable == "function" || typeof variable.name != "undefined") ? variable.name : ''));
+    };
+    var _isUndefined = function(variable) {
```

Comment:
I would suggest using lodash helper functions here e.g. https://lodash.com/docs/4.17.15#isBoolean instead of defining your own functions here.